config/task.yaml

```diff
@@ -0,0 +1,68 @@
+# Copyright 2022 Chainguard, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: apko
+spec:
+  description: "An apko task for building images."
+  params:
+    - name: dev.mink.sources.bundle
+      description: A self-extracting container image of source
+    - name: dev.mink.images.target
+      description: Where to publish an image.
+    - name: path
+      description: The bundle-relative path to the config file to build.
+      default: ./config.yaml
+
+  results:
+    - name: IMAGES
+      description: The list of images produced by this task (for chains)
+    - name: dev.mink.images.digest
+      description: The digest of the resulting image.
+
+  steps:
+    - name: extract-bundle
+      image: $(params["dev.mink.sources.bundle"])
+      workingDir: /workspace
+
+    - name: build-image
+      image: ko://chainguard.dev/apko/cmd/apko
+      workingDir: /workspace
+      args:
+      - build
+      - $(params["path"])
+      - $(params["dev.mink.images.target"])
+      # TODO(mattmoor): Move this to an emptyDir volume
+      - output.tar.gz
+
+    - name: push-image
+      # This is a version of "crane" with k8schain linked as
+      # a universal credential helper (for workload identity).
+      image: gcr.io/go-containerregistry/krane
+      workingDir: /workspace
+      args:
+      - push
+      - --image-refs=/tekton/results/IMAGES
+      - output.tar.gz
+      - $(params["dev.mink.images.target"])
+
+    # mink wants just the digest in its result, but the above produces
```

Comment:
Should mink take the Tekton form if given? That seems to be the thing we're recommending.

Reply:
> That seems to be the thing we're recommending

By "recommending" I assume you mean "The awful spelunking Chains does today", or is there something else?

... it was my understanding from the way the discussion on https://github.com/tektoncd/community/pull/504 went that Tekton has no such official "recommendations" today (this is what I was trying to fix, but I gave up).

Reply:
Right, yeah, chains' spelunking.

Whether or not it's a documented recommended format, yet, that seems to be the way we're recommending tools produce their output. It seems like a useful thing for mink to get on board with (and it strengthens the case for making it a documented recommendation)

Reply:
I can send a PR for mink when I get a chance.

Reply:
@imjasonh I'd MUCH rather see us do something better in Chains (and Tekton in general), and I don't think that ship has sailed.  Part of my desire for a stronger convention upstream for these things is that when matching Task interface characteristics, domain-scoping makes things (relatively) unambiguous.  I don't think `IMAGES` meets that bar.